mage_ai/data_preparation/shared/secrets.py

```diff
@@ -1,38 +1,70 @@
 import os
-from typing import List
+import uuid
+from typing import List, Tuple
 
 from cryptography.fernet import Fernet, InvalidToken
 
+from mage_ai.orchestration.constants import Entity
 from mage_ai.settings.repo import get_data_dir, get_repo_path
 
 DEFAULT_MAGE_SECRETS_DIR = 'secrets'
 
 
-def get_secrets_dir():
-    return os.path.join(
-        get_data_dir(), DEFAULT_MAGE_SECRETS_DIR)
+def get_secrets_dir(
+    entity: Entity,
+    project_uuid: str = None,
+    pipeline_uuid: str = None,
+) -> str:
+    secrets_dir = os.path.abspath(os.path.join(get_data_dir(), DEFAULT_MAGE_SECRETS_DIR))
+
+    if entity == Entity.GLOBAL:
+        return secrets_dir
+    elif entity == Entity.PROJECT:
+        return os.path.join(secrets_dir, 'projects', project_uuid)
+    elif entity == Entity.PIPELINE:
```

Comment:
Would pipeline renaming break the secrets?

Reply:
good point

Reply:
I don't have a good solution now. I think we can mention this issue in docs for now and ask user to manually rename the folder in prod.